wiring/inc/spark_wiring_system.h

```diff
@@ -764,6 +772,16 @@ class SystemClass {
 
 #endif // HAL_PLATFORM_POWER_MANAGEMENT
 
+    int setControlRequestFilter(system_control_req_filter* filter) {
```

Comment:
This needs to take a wrapper around `system_control_req_filter`. A suggestion is to take a similar approach to sleep API for example with the class that allows chaining `.call(123).call(345)` etc.

Reply:
+ some additional overloads that allow to pass an array/vector of ids.

Reply:
I gave this a bit more thought and now the API from a first glance looks a bit confusing. Does the filter specify enabled or disabled requests? I think we should make this more explicit. Let's do something along the lines of :
```c++
enum class SystemControlRequestAclAction {
    NONE = 0,
    ACCEPT = 1,
    DENY = 2
};
class SystemControlRequestAcl {
public:
    SystemControlRequestAcl(ctrl_request_type type, SystemControlRequestAclAction action);

    ctrl_request_type type() const;
    SystemControlRequestAclAction action() const;

private:
    ctrl_request_type type_;
    SystemControlRequestAclAction action_;
};

int setControlRequestFilter(SystemControlRequestAclAction defaultAction, Vector<SystemControlRequestAcl> filters);
```

With this you can do the following:
```c++
// Accept by default, deny some
System.setControlRequestFilter(SystemControlRequestAclAction::ACCEPT, { {CTRL_REQUEST_RESET, SystemControlRequestAclAction::DENY} });

// Deny by default, accept some
System.setControlRequestFilter(SystemControlRequestAclAction::DENY, { {CTRL_REQUEST_RESET, SystemControlRequestAclAction::ACCEPT} });
```

The appropriate system C APIs/structs will need to be added as well.

Reply:
@keeramis